packages/gatsby/src/schema/__tests__/run-query.js

```diff
@@ -360,6 +360,7 @@ describe(`Filter fields`, () => {
         hair: { eq: needle },
       })
 
+      expect(result?.length ?? 0).toEqual(new Set(result ?? []).size) // result should contain unique elements
```

Comment:
Maybe have `runFilter` to always return an array to avoid excessive optional chaining?

Reply:
I definitely want to do that once we can break majors. As it stands, we'll have to live with this empty array vs null result.

Reply:
I meant `runFilter` wrapper in tests here (not in the code itself): https://github.com/gatsbyjs/gatsby/blob/aa9e0be0938dd5a0bc9cc936ea664f5df3073172/packages/gatsby/src/schema/__tests__/run-query.js#L337-L344